src/sentry/stacktraces/processing.py

```diff
@@ -489,6 +490,35 @@ def dedup_errors(errors):
     return rv
 
 
+def suspected_console_errors(mechanism, error_type, frames):
+    def is_suspicious_frame(frame) -> bool:
+        function = frame.get("function", None)
+        filename = frame.get("filename", None)
+        return function == "?" and filename == "<anonymous>"
+
+    def has_suspicious_frames(frames) -> bool:
+        if len(frames) == 2 and is_suspicious_frame(frames[0]):
+            return True
+        return all(is_suspicious_frame(frame) for frame in frames)
+
+    if (
+        not frames
+        or not mechanism
+        or mechanism.get("type") != "onerror"
+        or mechanism.get("handled")
+    ):
+        return False
+
+    has_short_stacktrace = len(frames) <= 2
+    is_suspicious_error = error_type.lower() in [
+        "syntaxerror",
+        "referenceerror",
+        "typeerror",
+    ]
```

Comment:
Do you have some Sentry queries representing each of these so we can inspect them? Or an explanation of what each one represents?

Reply:
Here's some search queries. You can set status to ignored to see more example issues.

[TypeError](https://sentry.io/organizations/sentry/issues/?query=is%3Aunresolved+error.mechanism%3Aonerror+error.handled%3Afalse+error.type%3ATypeError+stack.filename%3A%3Canonymous%3E&statsPeriod=14d)
[ReferenceErrpr](https://sentry.io/organizations/sentry/issues/?query=is%3Aunresolved+error.mechanism%3Aonerror+error.handled%3Afalse+error.type%3AReferenceError+stack.filename%3A%3Canonymous%3E&statsPeriod=14d)
[SyntaxError](https://sentry.io/organizations/sentry/issues/?query=is%3Aunresolved+error.mechanism%3Aonerror+error.handled%3Afalse+error.type%3ASyntaxError+stack.filename%3A%3Canonymous%3E&statsPeriod=14d)